src/pycountry/tests/test_general.py

```diff
@@ -317,7 +317,7 @@ def test_add_entry():
     )
 
     country = pycountry.countries.get(alpha_2="XK")
-    assert isinstance(country, pycountry.countries.data_class)
+    assert isinstance(country, pycountry.countries.factory)
 
```

Comment:
@terencehonles additionally I'd like to propose amending the `test_general.py` file with the following tests as part of this merge. While I know `mypy` handles type checking in future iterations I'd love to avoid these classes getting changed to something other than expected / anticipated. 

Would be interested in your thoughts as well on this.

``` python
def test_database_instance_of_language():
    assert isinstance(pycountry.languages, pycountry.Languages)


def test_database_instance_of_country():
    assert isinstance(pycountry.countries, pycountry.ExistingCountries)


def test_database_instance_of_subdivision():
    assert isinstance(pycountry.subdivisions, pycountry.Subdivisions)


def test_database_instance_of_script():
    assert isinstance(pycountry.scripts, pycountry.Scripts)


def test_instance_of_currency():
    assert isinstance(list(pycountry.currencies)[0], pycountry.Currency)


def test_instance_of_language():
    assert isinstance(list(pycountry.languages)[0], pycountry.Language)


def test_instance_of_country():
    assert isinstance(list(pycountry.countries)[0], pycountry.Country)


def test_instance_of_script():
    assert isinstance(list(pycountry.scripts)[0], pycountry.Script)


def test_instance_of_currency():
    assert isinstance(list(pycountry.currencies)[0], pycountry.Currency)
```

Reply:
It looks like a number of these already existed. I used `pytest.mark.parameterize` to collapse these somewhat duplicative tests and added the `isinstance` check on the first item as you suggested.

Reply:
> While I know `mypy` handles type checking in future iterations I'd love to avoid these classes getting changed to something other than expected / anticipated.

Is mypy configured to run as a check? I'm not 100% sure it is, and that would be a good thing to enable, but I didn't spend ~too much~ any time making sure it does.

Reply:
@terencehonles love the `pytest.mark.parameterize` use to collapse these test. In terms of the `mypy` check it isn't included in the Makefile currently but #183 should address that. It is part of the .github/workflows on main under the mypy-check.yml file.

Reply:
cool